File: mobile/lib/features/notifications/domain/entities/notification.dart
class AppNotification {
  const AppNotification({
    required this.id,
    required this.type,
    required this.channel,
    required this.title,
    required this.body,
    required this.status,
    required this.createdAt,
  });

  final String id;
  final String type;
  final String channel;
  final String title;
  final String body;
  final String status;
  final String createdAt;
}

File: backend/services/core/internal/domain/notification.go
package domain

import "time"

// NotificationChannel selects the delivery path for an event.
type NotificationChannel string

const (
	ChannelEmail NotificationChannel = "email"
	ChannelPush  NotificationChannel = "push"
	ChannelBoth  NotificationChannel = "email+push"
)

// Notification is an in-app + delivery record.
type Notification struct {
	ID        string              `json:"id"`
	UserID    string              `json:"userId"`
	Type      string              `json:"type"`
	Channel   NotificationChannel `json:"channel"`
	Title     string              `json:"title"`
	Body      string              `json:"body"`
	Status    string              `json:"status"` // queued | sent | failed
	CreatedAt time.Time           `json:"createdAt"`
}

// FCMToken records a device token for push delivery.
type FCMToken struct {
	ID        string    `json:"id"`
	UserID    string    `json:"userId"`
	Token     string    `json:"token"`
	Platform  string    `json:"platform"`
	CreatedAt time.Time `json:"createdAt"`
}
